rtrlib/rtrlib
# BGPsec support for RTRlib

### Contribution description


This PR offers BGPsec validation and signing routines. It extends the current API with functions for
* validating a BGPsec path
* generating a signature for a BGPsec path
* getter functions for BGPsec version and algorithm suites.

The implementation relies on OpenSSL and is compatible with version 1.0 and 1.1. BGPsec functionality is optional. RTRlib can still be build without it. BGPsec features are enabled by default, so no further arguments must be passed to `cmake`. There are however options to disable BGPsec, see the README for more information on this. When building in `Debug` mode, RTRlib will output more information on validation procedures.

```bash
cmake -D CMAKE_BUILD_TYPE=Debug
```

The usage of the API is similar to that of prefix validation. First, RTRlib structures have to be filled with the necessary path information. These structures are then passed to the validation/signing function. The validation function will either return valid/not valid or an appropriate error code. The signing function will generate a signature and its length, or again an error code.

This code has been reviewed by @mroethke, @smlng and @reuteran so far. I am open to more review and suggestions.

### Testing procedure


When RTRlib is compiled with BGPsec support, simply run

```bash
make test
```

to run all tests, including `test_bgpsec`. To view the results of every validation, run the test manually. This implies that RTRlib was build in `Debug` mode. Otherwise, there is not much to see.

### Issues/PRs references
